src/averageAges.js

```diff
@@ -37,7 +52,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const peopleFiltered = people.filter((person) => withChildren
+    ? people.some((child) => child.mother === person.name)
+    : person.sex === 'f');
+
+  const womenAge = peopleFiltered.reduce((accumulator, currentValue) =>
+    (accumulator + (currentValue.died - currentValue.born)), 0);
+
+  const womenAverageAge = womenAge / peopleFiltered.length;
+
+  return womenAverageAge;
```

Comment:
```suggestion
  const womenAge = peopleFiltered.reduce((accumulator, currentValue) =>
    (accumulator + (currentValue.died - currentValue.born)), 0);

  return womenAge / peopleFiltered.length;
```
you are doing the same at least twice (or even 3 times) - move it to a separate function